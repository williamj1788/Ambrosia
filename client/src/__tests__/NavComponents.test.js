import React from 'react';
import { shallow } from 'enzyme';
import { 
    Navigation,
    NavContainer,
    NavLinkContainer,
    HamburgerDropdown,
    NavLinkWrapper,
 } from '../components/NavComponents';
import toJson from 'enzyme-to-json';
import { Redirect } from 'react-router-dom';


describe('<Navigation /> Snapshop Tests',() => {
    test('renders nav with fixed class when fixed prop is true', () => {
        const component = shallow(<Navigation fixed={true} />);
        expect(toJson(component)).toMatchSnapshot();
    });
    
    test('renders nav with no class when fixed prop is false', () => {
        const component = shallow(<Navigation fixed={false} />);
        expect(toJson(component)).toMatchSnapshot();
    });
});

describe('<Navigation /> snapshop Test', () => {
    test('renders div with transparent background when transparent prop is true', () => {
        const component = shallow(<NavContainer transparent={true} />); 
        expect(toJson(component)).toMatchSnapshot(); 
    });

    test('renders div with no transparent background when transparent prop is false', () => {
        const component = shallow(<NavContainer transparent={false} />); 
        expect(toJson(component)).toMatchSnapshot(); 
    });
});

describe('<NavLinkContainer /> snapshop Test', () => {
    test('renders div with correct classes and inline-styles', () => {
        const props = {
            className: 'dummy',
            style: {
                color: 'blue'
            }
        }
        const component = shallow(<NavLinkContainer {...props} />); 
        expect(toJson(component)).toMatchSnapshot(); 
    });
});

describe('<HamburgerDropdown /> snapshop Test', () => {
    test('renders <NavLinkContainer /> with correct style hieght when open prop is true', () => {
        const component = shallow(<HamburgerDropdown open={true} />); 
        expect(toJson(component)).toMatchSnapshot(); 
    });

    test('renders <NavLinkContainer /> with correct style hieght when open prop is false', () => {
        const component = shallow(<HamburgerDropdown open={false} />); 
        expect(toJson(component)).toMatchSnapshot(); 
    });
});

describe('<NavLinkWrapper /> snapshop Test', () => {
    test('renders img if icon prop is present', () => {
        const component = shallow(<NavLinkWrapper icon={'mock icon'} />); 
        expect(component.find('img')).toHaveLength(1); 
    });

    test('renders no img if icon prop is not present', () => {
        const component = shallow(<NavLinkWrapper />); 
        expect(component.find('img')).toHaveLength(0); 
    });

    test('renders span with correct style if TO prop is not present', () => {
        const component = shallow(<NavLinkWrapper />); 
        expect(component.find('span').prop('style')).toHaveProperty('cursor', 'pointer');
    });
});