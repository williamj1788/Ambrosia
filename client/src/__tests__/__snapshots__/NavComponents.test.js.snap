// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HamburgerDropdown /> snapshop Test renders <NavLinkContainer /> with correct style hieght when open prop is false 1`] = `
<NavLinkContainer
  className="HamburgerDropdown"
  style={
    Object {
      "height": "0",
      "paddingBottom": "0",
    }
  }
/>
`;

exports[`<HamburgerDropdown /> snapshop Test renders <NavLinkContainer /> with correct style hieght when open prop is true 1`] = `
<NavLinkContainer
  className="HamburgerDropdown"
  style={
    Object {
      "height": "200px",
    }
  }
/>
`;

exports[`<NavLinkContainer /> snapshop Test renders div with correct classes and inline-styles 1`] = `
<div
  className="dummy"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <NavLinkWrapper
    active={true}
    exact={true}
    to="/"
  >
    Home
  </NavLinkWrapper>
  <NavLinkWrapper
    active={true}
    strict={true}
    to="/menu"
  >
    Menu
  </NavLinkWrapper>
  <NavLinkWrapper
    active={true}
    to="/meet"
  >
    Meet The Chiefs
  </NavLinkWrapper>
  <NavLinkWrapper
    icon="CartIcon.png"
  >
    Cart
  </NavLinkWrapper>
  <NavLinkWrapper
    icon="UserIcon.png"
    to="/login"
  >
    Login
  </NavLinkWrapper>
</div>
`;

exports[`<Navigation /> Snapshop Tests renders nav with fixed class when fixed prop is true 1`] = `
<nav
  className="fixed"
/>
`;

exports[`<Navigation /> Snapshop Tests renders nav with no class when fixed prop is false 1`] = `<nav />`;

exports[`<Navigation /> snapshop Test renders div with no transparent background when transparent prop is false 1`] = `
<div
  className="navbar"
  style={Object {}}
>
  <div
    className="container"
  />
</div>
`;

exports[`<Navigation /> snapshop Test renders div with transparent background when transparent prop is true 1`] = `
<div
  className="navbar"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="container"
  />
</div>
`;
